test/pathofthetemplarshard.ts

```diff
@@ -0,0 +1,125 @@
+import { ethers } from "hardhat";
+import { TypedDataDomain, TypedDataField, TypedDataSigner } from "@ethersproject/abstract-signer";
+import { expect } from "chai";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import {
+  PathOfTheTemplarShard, PathOfTheTemplarShard__factory,
+  Relic, Relic__factory,
+  Shards, Shards__factory,
+} from "../typechain";
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import path from "path";
+
+type MySigner = SignerWithAddress & TypedDataSigner;
+
+let owner: MySigner;
+let add2: MySigner;
+let add3: MySigner;
+let add4: MySigner;
+let pathOfTheTemplarShard: PathOfTheTemplarShard;
+let relic: Relic;
+let shards: Shards;
+let SHARD_ID: number[] = [2, 3, 4, 5, 6];
+let ENCLAVE: string[] = [
+    "",
+    "chaosEnclave",
+    "mysteryEnclave",
+    "logicEnclave",
+    "structureEnclave",
+    "orderEnclave"
+];
+
+describe("PathOfTheTemplarShard", async () => {
+
+    before( async () => {
+        [owner, add2, add3, add4] = await ethers.getSigners();
+    });
+
+    async function setup() {
+        relic = await new Relic__factory(owner).deploy();
+        shards = await new Shards__factory(owner).deploy();
+        pathOfTheTemplarShard = await new PathOfTheTemplarShard__factory(owner).deploy();
+
+        return {
+            relic,
+            shards,
+            pathOfTheTemplarShard,
+        }
+    }
+
+    beforeEach( async () => {
+        ({
+            relic,
+            shards,
+            pathOfTheTemplarShard,
+        } = await loadFixture(setup));
+    });
+
```

Comment:
some tests are missing. check test coverage and add tests for view functions too